cypress/integration/listingFlows/deleteListing.spec.js

```diff
@@ -0,0 +1,31 @@
+describe('Delete listing', () => {
+  beforeEach(() => {
+    cy.testSetup();
+    cy.viewport('macbook-16');
+
+    cy.fixture('users/adminUser.json').as('user');
+
+    cy.get('@user').then((user) => {
+      cy.loginUser(user).then(() => {
+        cy.createListing({
+          title: 'Test Listing',
+          content: `This is a test listing's contents.`,
+        }).then(() => {
+          cy.visit('/listings/dashboard');
+        });
+      });
+    });
+  });
+
+  it('deletes a listing', () => {
+    cy.findByText('Delete').click();
+
+    cy.findByRole('main').within(() => {
+      cy.get('button', { name: /^Delete$/i }).click();
+    });
```

Comment:
```suggestion
    cy.findByRole('main').findByRole('button', { name: /^Delete$/i }).click();
```

As there's only one action we want to complete within the `main`, we can just chain it off the first `findBy...` without using `within` 🙂

Reply:
TIL!